addons/coupon/models/coupon_program.py

```diff
@@ -53,6 +53,11 @@ class CouponProgram(models.Model):
         help="Validity duration for a coupon after its generation")
     total_order_count = fields.Integer("Total Order Count", compute="_compute_total_order_count")
 
+    @api.constrains('rule_min_quantity', 'reward_type')
```

Comment:
there is no field "rule_min_quantity" on this object.

you probably need to use rule_id.rule_min_quantity, same bellow in the condition